### ISSUE TITLE
Iocation info

### ISSUE THREAD

**Issue by abdul-alhasany** [pfaciana/parse-docblocks#1]
Hi 
Is it possible to add location info to each item (description, tags .. etc).

Location info could be like:
```js
start:{
    line: startingLine,
    col: startingCol,
    offset: startingOffset
},
end: {
    line: endingLine,
    col: endingCol,
    offset: endingOffset
}
```

It can also be added to configuration object so it can be enabled/disabled as needed in the output.

**Comment by pfaciana** [comment]
Hi! thanks for asking.

I should be able to add line numbers fairly easily. Column/offset might be trickier as I am stripping out whitespace from all the lines right at the start of the application, and the rest of the application is dependent on that not being there.

I will say that AFAIK different comment parts can not share the same line (at least not conventionally) and my application does not support that. Line delimitation is an important part of the parsing. Can I ask your use case, and if knowing the line numbers (start and end) is enough since the entire line (or lines) would be associated with that piece of the comments?

Thanks!

**Comment by abdul-alhasany** [comment]
Thanks for the reply. Having line numbers is a good start but it would much more help to have the column/offset number.

I am working on a PHP linter/fixer. Part of its functionality is to lint docblocks and fix some errors. The offset is needed to manipulate the string at a certain point. Currently I am splitting the lines and use indexOf/regex to find the offset position.

**Comment by pfaciana** [comment]
Hi, I've been staring at this for the last few days and the issue I keep running into is that I am manipulating, hashing and running recursion to the data throughout the application. By the time I get to building out the object I cannot trace the output back to the original input (unless the use case is very basic). I've sort of run out of clever hacks and tricks to try, but I'm open to suggestions. Do you have any thoughts?

A little background on this repo, I created this to help me automate building user documentation (for use on wikis, like for github wikis, and building webpages docs, etc). Would it be possible to take the json, replace/fix what's needed and convert it back to comments? I know that's not what you're hoping for, but its how I wrote the code to used. I'm sorry I haven't been able to come up with a stable solution yet, but I'll continue to see if I can come up with an answer. Please let me know your thoughts.

**Comment by abdul-alhasany** [comment]
Thanks for looking into this.
I think the repo is doing its job but it might need an upgrade. I believe it will be a good idea to re-write the code base. We can leverage some of the existing code into the new version.

If you are willing for this plan I made a proof of concept using vite + typescript. You can find the repo here:
https://github.com/kalimah-apps/docblock-parser

It does basic things for now but I think you will get the idea. It provides position data for tags including line, col and offset for start and end. I think it is a good starting point to develop into a more robust code with support for more cases, tags and edge cases.

Let me know what you think.

**Comment by pfaciana** [comment]
I agree in order for the repo to add additional functionality, it needs to be re-written. And I'm happy to assist in a new repo that leverages some of the functions and concepts I created in the original repo. The obstacle I face is availability. My schedule is usually booked full and I work on these mini-projects in the short bursts of downtime I have, but it's timing is very unpredictable. I very much support what you're asking, but I don't want to give the impression I would be able to dedicate a lot of time to re-writing this in the near future. My schedule just won't allow it. However, I can make myself available as a resource. If I have time, I'll help, but if I don't, I can only answer questions or point you in the direction that I took.
